lib/search/best_bets_checker.rb

```diff
@@ -47,52 +43,35 @@ def fetch
     #
     # If a link occurs at multiple positions, it will only occur in the group for
     # the highest position it occurred at.
+    # Where high means the smallest positional value.
     def combine_best_bets(bets)
-      by_position = bets.map do |bet|
-        [bet["position"], bet["link"]]
+      bets
+      .map { |bet| [bet['position'], bet['link']] }
+      .sort
+      .uniq { |_, link| link }
+      .each_with_object(Hash.new) do |(position, link), result|
+        result[position] ||= []
+        result[position] << link
       end
-      by_position.sort!
-
-      combined = Hash.new
-      seen = Set.new
-      by_position.each do |bet|
-        position = bet[0]
-        link = bet[1]
-        if seen.include? link
-          next
-        end
-        seen.add link
-        (combined[position] ||= []) << link
-      end
-      combined
     end
 
     def combine_worst_bets(bets)
-      combined = Set.new
-      bets.each do |bet|
-        combined.add bet["link"]
-      end
-      combined.to_a
+      bets.map { |bet| bet['link'] }.uniq
```

Comment:
The `.to_a` was turning it back into an array so this just skips the set altogether.